### PR TITLE
UI improvements

### PR DESCRIPTION
**Pull request by dmitriim**

### CODE REVIEW

File: tests/external/rule_conditions_test.php [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+use tool_cohortmanager\condition;
+use tool_cohortmanager\rule;
+use tool_cohortmanager\tool_cohortmanager\condition\cohort_membership;
+use tool_cohortmanager\tool_cohortmanager\condition\user_profile;
+
+defined('MOODLE_INTERNAL') || die();
+
+global $CFG;
+require_once($CFG->dirroot . '/lib/externallib.php');
+require_once($CFG->dirroot . '/webservice/tests/helpers.php');
```
Comment (golenkovm) [inline tests/external/rule_conditions_test.php:29]: I like the idea of moving require calls into `setUpBeforeClass()` method and removing `defined('MOODLE_INTERNAL') || die();`. Not sure if it's advised by Moodle coding standards, so will leave it to you.

File: tests/external/rule_conditions_test.php [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+ *
+ * @package    tool_cohortmanager
+ * @author     Dmitrii Metelkin <dmitriim@catalyst-au.net>
+ * @copyright  2022 Catalyst IT
+ * @license    http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+class rule_conditions_test extends externallib_advanced_testcase {
+
+    /**
+     * Test exception if rule is not exist.
```
Comment (golenkovm) [inline tests/external/rule_conditions_test.php:42]: Should `@cover` tag be specified?
```suggestion
     * Test exception if rule is not exist.
     * @covers ::get_conditions
```

File: tests/external/rule_conditions_test.php [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+
+        $this->setAdminUser();
+        $this->expectException(\invalid_parameter_exception::class);
+        $this->expectExceptionMessage('Rule does not exist');
+
+        rule_conditions::get_conditions(777);
+    }
+
+    /**
+     * Test required permissions.
```
Comment (golenkovm) [inline tests/external/rule_conditions_test.php:55]:
```suggestion
     * Test required permissions.
     * @covers ::get_conditions
```

File: tests/external/rule_conditions_test.php [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+        $this->setUser($user);
+
+        $this->expectException(\required_capability_exception::class);
+        $this->expectExceptionMessage('Sorry, but you do not currently have permissions to do that (Manage rules).');
+
+        rule_conditions::get_conditions(777);
+    }
+
+    /**
+     * Test getting conditions for empty rule.
```
Comment (golenkovm) [inline tests/external/rule_conditions_test.php:69]:
```suggestion
     * Test getting conditions for empty rule.
     * @covers ::get_conditions
```

File: tests/external/rule_conditions_test.php [hunk 5]
```diff
@@ -0,0 +1,155 @@
...
+        $rule = new rule(0, (object)['name' => 'Test rule 1']);
+        $rule->save();
+
+        $conditions = rule_conditions::get_conditions($rule->get('id'));
+        $this->assertIsArray($conditions);
+        $this->assertEmpty($conditions);
+    }
+
+    /**
+     * Test getting conditions for aa rule.
```
Comment (golenkovm) [inline tests/external/rule_conditions_test.php:84]:
```suggestion
     * Test getting conditions for aa rule.
     * @covers ::get_conditions
```

File: tests/external/rule_conditions_test.php (hunk 1 above)
Comment (dmitriim) [reply tests/external/rule_conditions_test.php:29]: My personal preference is to have require_once section in the beginning of the file.

**Review by golenkovm — APPROVED** [review]
LGTM :+1:

**Pull request merged**